### PR TITLE
ADD querystring flags to force h264 and dtx

### PR DESCRIPTION
**Pull request by ggarber**

### CODE REVIEW

File: src/js/webrtc-intercept.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+(function() {
+
+  function getParameterByName(name, url) {
```
Comment (aullman) [inline src/js/webrtc-intercept.js]: This function is repeated in ot-intercept.js let's write it once and require it in.

Comment (ggarber) [inline src/js/webrtc-intercept.js]: ok, I understood that you wanted to be able to remove it easily as soon as we had a good way to configure codecs in opentok.js so i didn't make a lot of effort on this.  Will do now.

**Comment by aullman** [comment]
@ggarber I'm no sure if I'm not testing things correctly but I can't get H264 to work. I'm just testing locally going to localhost:5000/room?h264=true and when I use Chrome to Chrome and look in webrtc-internals I just see normal VP8. 

image


When I do Firefox to Chrome though I get VP9??

image

Am I missing something? Did I break something in my refactor?

**Comment by aullman** [comment]
@ggarber ok, I added a simple test that just makes sure that you can successfully connect and subscribe when you turn on h264.

**Comment by ggarber** [comment]
@aullman 
1) It is what I tried to (badly) explain in slack.  You need to load part of the hack before opentok.js and the other part after opentok.js.  That's the reason why I had 2 scripts in webpack configuration.   I'll try to fix it.

2) H264 should work with fake devices.  Devices and codecs should be independent.    I wouldn't waste lot of time with this though.  It is very hacky no matter what and not business critical.

3) Firefox to Chrome VP9 is very surprising.  I will review that.

**Comment by ggarber** [comment]
@aullman 

1. It was not working because I test it with Chrome Canary and you probably used Chrome and apparently the payload types have changed.  I fixed it to support both version now.

2. VP9 in Firefox was happening because your firefox version doesn't support H264 (it happens to me too if use Stable instead of Nightly).  I fixed to make the rule more strict, now it will never use VP9.

3. I think tests should work fine with fake devices.   I'm not sure how much effort we should put in this though if it is very hacky no matter what.  Maybe it is better to work first on exposing in opentok.js the APIs needed to configure the codec list and/or manipulate the SDPs properly.  But I let you decide :)

**Comment by aullman** [comment]
Thanks @ggarber looking good. I also added the video codec to the stats drop down using the QOS event from the subscriber. I also added a quick test that verifies that the stats show the H264 codec if you pick H264. It only works in Chrome because Firefox still doesn't have the codec name in the stats.

File: src/js/h264/modify-sdp.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  if (OrigPeerConnection) {
+    var newPeerConnection = function(config, constraints) {
+        console.log('PeerConnection created with config', config);
+
+        var pc = new OrigPeerConnection(config, constraints);
+        var origSetRemoteDescription = pc.setRemoteDescription.bind(pc);
+        pc.setRemoteDescription = function(sdp, success, failure) {
+          console.log('Intercept setRemoteDescription');
+          if (dtx) {
+            sdp.sdp = sdp.sdp.replace('useinbandfec=1', 'useinbandfec=1;usedtx=1');
```
Comment (odivorra) [inline src/js/h264/modify-sdp.js]: @aullman , @ggarber  are we guys mangling this without telling the remote receiver in the offer?

Comment (odivorra) [inline src/js/h264/modify-sdp.js]: can this be a problem?

**Pull request merged**